### DISCUSSION TITLE
Native Node.js SQLite adapter

### DISCUSSION THREAD

**Discussion by darky** [mikro-orm/mikro-orm#5886] [Ideas]
Since 22.5.0 version Node.js has own built-in SQLite module. 
Suggestion to add adapter to MikroORM for native Node.js SQLite adapter. Maybe it's will be more performant than `better-sqlite3`. At least, it's allows to reduce amount of dependencies (`better-sqlite3`)

**Comment by B4nan** [comment]
Creating new sqlite drivers is now trivial, 99% of the code is already abstracted and extensible, so I'll leave this up for the community, at least for the time being, especially since this new native node sqlite support is still experimental (with stability 1).

Technically all you need is a custom knex dialect like this one:

https://github.com/mikro-orm/mikro-orm/blob/master/packages/knex/src/dialects/sqlite/BetterSqliteKnexDialect.ts#L7

And use that through the connection class:

https://github.com/mikro-orm/mikro-orm/blob/master/packages/better-sqlite/src/BetterSqliteConnection.ts#L6

Rest can be all just empty extensions of the required classes (platform and driver basically).

**Reply by darky** [reply]
Seems that **Knex** should to have support of native Node.js SQLite module before?

**Reply by B4nan** [reply]
Knex is just a wrapper around libraries like better-sqlite3, all you need is a knex dialect, you can reuse the rest as long as it has the same API - which is most likely the case, either the sqlite3 or better-sqlite3 dialects should be compatible. Look at the libsql driver, knex also doesn't support it natively, we just create the dialect here and pass it explicitly, its the very same story.

**Comment by piotr-cz** [comment]
It's possible since v7.0 without custom database adapter and it's described here: Docs > Integrations > [Using node:sqlite](https://mikro-orm.io/docs/7.0/usage-with-sqlite#using-nodesqlite).

Though I don't see any difference in performance to `@mikro-orm/sqlite` (v7, v6) or `@mikro-orm/better-sqlite` (v6)
